### PR TITLE
Add remoteAdbHost to desired caps

### PR DESCRIPTION
**Pull request by serggl**
Promote `remoteAdbHost` as a desired capability (https://github.com/appium/appium/issues/8564)

### CODE REVIEW

File: lib/android-helpers.js [hunk 1]
```diff
@@ -175,8 +176,8 @@ helpers.getDeviceInfoFromCaps = async function (opts = {}) {
 };
 
 // returns a new adb instance with deviceId set
-helpers.createADB = async function (javaVersion, udid, emPort, adbPort, suppressKillServer) {
-  let adb = await ADB.createADB({javaVersion, adbPort, suppressKillServer});
+helpers.createADB = async function (javaVersion, udid, emPort, remoteAdbHost, adbPort, suppressKillServer) {
+  let adb = await ADB.createADB({javaVersion, remoteAdbHost, adbPort, suppressKillServer});
```
Comment (dpgraham) [inline lib/android-helpers.js]: Can we just move `remoteAdbHost` after `suppressKillServer` to make it a non-breaking change?

**Comment by imurchie** [comment]
To mitigate that, and because the parameter list is getting pretty long, it could be changed into an options hash.

File: lib/android-helpers.js (hunk 1 above)
Comment (serggl) [inline lib/android-helpers.js]: @dpgraham agree. Just pushed a change

**Comment by dpgraham** [comment]
@imurchie Do you want just the last variable to be a hash? Like this:

`helpers.createADB = async function (javaVersion, udid, emPort, adbPort, suppressKillServer, {remoteAdbHost})`

That would maintain backward compatibility.

Otherwise we make all the args a hash like this

`+helpers.createADB = async function ({javaVersion, udid, emPort, adbPort, suppressKillServer, remoteAdbHost})`

And then we make this a major version bump and refactor any code that uses this to the new format.

**Comment by imurchie** [comment]
I think the change that was made, suggested by @dpgraham, is sufficient.

**Comment by imurchie** [comment]
Published in `appium-android-driver@1.21.0`.

**Pull request merged**